golang/cosmos/x/swingset/client/cli/tx.go

```diff
@@ -42,10 +42,34 @@ func GetTxCmd(storeKey string) *cobra.Command {
 	return swingsetTxCmd
 }
 
+// parseInput handles an input argument that may refer to a file.
+// If the input does not start with a @, it is returned as-is.
+// The slice after the @ represents the filename. If the filename is -
+// a single line from stdin is returned, otherwise the full content
+// of the file is returned.
+func parseInput(input string) (output string, err error) {
```

Comment:
Nit: I dislike using named return values when the return statements don't use them in all cases - the semantics are subtle, especially if there's anything remotely tricky like a `defer`. Consider just using unnamed `(string, error)` and then add the one declaration of `var output string` before line 58.